plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -624,7 +624,7 @@ func PodFitsResources(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.No
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
 	for _, req := range nodeSelectorTerms {
-		nodeSelector, err := v1helper.NodeSelectorRequirementsAsSelector(req.MatchExpressions)
```

Comment:
Tightening the validation may introduce backward compatibility. I'm thinking to check it at least once as you said :).